benchmarks/piazza/piazza.rs

```diff
@@ -0,0 +1,332 @@
+            .long("dcfg")
+            .possible_values(&["single", "peruser"])
+            .takes_value(true)
+            .default_value("single")
+            .help("Domain assignment configuration"))
+        .arg(Arg::with_name("benchmark")
+            .possible_values(&["write", "migration"])
+            .takes_value(true)
+            .required(true)
+            .help("Benchmark configuration"))
+        .get_matches();
+
+
+    println!("Creating app...");
+    let mut app = Piazza::new();
+    println!("Done with app creation.");
+
+    let nusers = value_t_or_exit!(args, "nusers", i64);
+    let nclasses = value_t_or_exit!(args, "nclasses", i64);
+    let nposts = value_t_or_exit!(args, "nposts", i64);
+    let benchmark = args.value_of("benchmark").unwrap();
+    let domain_config = args.value_of("domain_config").unwrap();
+    let fanout = args.value_of("fanout").unwrap();
+    let csv = args.is_present("csv");
+
+    let class_putter = app.soup.get_mutator(app.class);
+    let user_putter = app.soup.get_mutator(app.user);
+    let taking_putter = app.soup.get_mutator(app.taking);
+    let post_putter = app.soup.get_mutator(app.post);
+
+    println!("Seeding...", );
+    populate_users(nusers, user_putter);
+    populate_classes(nclasses, class_putter);
+    populate_taking(nclasses, nusers, taking_putter, fanout);
+
+    match benchmark.as_ref() {
+        "migration" => {
+            for pid in 0..nposts {
+                post_putter.put(vec![
+                    pid.into(),
+                    (pid % nclasses).into(),
+                    (pid % nusers).into(),
+                    "post".into()
+                    ]);
+            }
+        },
+        "write" => {
+
+        }
+        _ => {
+            println!("wrong benchmark!");
+            return
+        }
+    }
+
+    println!("Finished seeding! Sleeping for 1 second...");
+    thread::sleep(time::Duration::from_millis(1000));
+
+
+    let mut times = Vec::new();
+    if csv {
+       File::create("out.csv").unwrap();
+    }
+
+    println!("Starting benchmark...");
+    for uid in 0..(nusers) {
+        let start;
+        let end;
+
+        match benchmark.as_ref() {
+            "migration" => {
+                start = time::Instant::now();
+                app.log_user(uid.into(), domain_config);
+
+                end = time::Instant::now().duration_since(start);
+            },
+            "write" => {
+                start = time::Instant::now();
+                for i in 0..1000 {
+                    post_putter.put(vec![i.into(), (i % nclasses).into(), (i % nusers).into(), "post".into()]);
+                }
+                end = time::Instant::now().duration_since(start);
+
+                thread::sleep(time::Duration::from_millis(1000));
+
+                app.log_user(uid.into(), domain_config);
+            },
+            _ => {
+                println!("wrong benchmark!");
+                return
+            }
+        };
```

Comment:
It avoids replicating the `app.log_user` call, which is nice to ensure we don't accidentally make them different somehow. The `end1` `end2` game is unfortunate though. I'm okay with either solution.